### PR TITLE
Fixinclexer

### PR DESCRIPTION
**Pull request by ptersilie**
Added some fixes for the incremental lexer. Most importantly this fixes the lookback attributes which weren't always correct.

### CODE REVIEW

File: lib/eco/inclexer/inclexer.py [hunk 1]
```diff
@@ -586,31 +584,38 @@ def relex(self, node):
...
 
         changed = False
-        node_before_changes = None # remember where we started relexing
-        for tokens, read in pairs:
-            # merge new tokens into the parse tree
-            if not node_before_changes:
-                node_before_changes = read[0].prev_term
-            if self.merge_pair(tokens, read):
-                changed = True
+        # remember the location just before we started relexing so we can
```
Comment (ltratt) [inline lib/eco/inclexer/inclexer.py]: Can we rephrase this (maybe as a couple of sentences?).

File: lib/eco/inclexer/inclexer.py [hunk 2]
```diff
@@ -886,27 +896,61 @@ def make_token(self, start, end, tokentype):
         if end == -1:
             end = sys.maxint
 
-        while i < end:
+        mtokens = []
+
+        for node in iter_tree(node):
+            if i >= end:
```
Comment (ltratt) [inline lib/eco/inclexer/inclexer.py:902]: Is it intentional that `i` will be set to `range(node.lookback)[-1]`.

Comment (ltratt) [reply lib/eco/inclexer/inclexer.py:902]: Ignore this comment -- it's completely wrong!

**Comment by ptersilie** [comment]
Addressed your comments.

File: lib/eco/inclexer/inclexer.py [hunk 3]
```diff
@@ -584,10 +584,10 @@ def relex(self, node):
                 return False
 
         changed = False
-        # remember the location just before we started relexing so we can
-        # afterwards update all lookback values even if we inserted nodes before
-        # the initial starting node or `node` has been moved into a multitext
-        # node
+        # We have to remember the location at which we started relexing. This
+        # allows us to properly updated all lookback values, even if nodes have
```
Comment (ltratt) [inline lib/eco/inclexer/inclexer.py]: "updated" -> "update"

Comment (ptersilie) [reply lib/eco/inclexer/inclexer.py]: Whoops. Fixed.

**Comment by ltratt** [comment]
Looks good. Squashing time!

**Comment by ptersilie** [comment]
Squashed

**Pull request merged**